services/forms/repo_form.go

```diff
@@ -591,7 +591,8 @@ type MergePullRequestForm struct {
 	MergeCommitID          string // only used for manually-merged
 	HeadCommitID           string `json:"head_commit_id,omitempty"`
 	ForceMerge             *bool  `json:"force_merge,omitempty"`
-	DeleteBranchAfterMerge bool   `json:"delete_branch_after_merge,omitempty"`
+	MergeWhenChecksSucceed bool
```

Comment:
I am confused...
Why are some fields tagged with json metadata, and some are not?
I think we should handle it unified: Either give everything the `json:"<name>,omitempty"` tag, or don't assign it at all.
So, for the newly added variable this would look like
```suggestion
	MergeWhenChecksSucceed bool `json:"merge_when_checks_succeed,omitempty"`
```